visio_gui.py

```
@@ -0,0 +1,537 @@
+                    files_data = [files_data]
+                elif not isinstance(files_data, list):
+                    files_data = []
+                self.root.after(0, self._scan_complete, files_data)
+            except json.JSONDecodeError as e:
+                self.root.after(0, lambda: messagebox.showerror("Error", f"Error parsing scan results: {str(e)}"))
+                print(f"JSON parsing error: {e}")
+                print(f"Raw output: {result.stdout}")
+                
+        except subprocess.TimeoutExpired:
+            self.root.after(0, lambda: messagebox.showerror("Error", "Scan timed out after 60 seconds."))
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)